### DISCUSSION TITLE
GitHub Enterprise Server 3.9; Self-Hosted Runner and Custom Action

### DISCUSSION THREAD

**Discussion by pulkownik1803** [community/community#85665] [Enterprise]
### Select Topic Area

Question

### Body

Hi, I have an issue with Action wrote by myself. I've use the official template to created action but unfortunately it won;t work in my environment.
Problem is when I've use **node20** in _action.yaml_ file - then I'm getting an error saying that this value is not allowed.

My environment:
GitHub Enterprise Server 3.9.6
Self-hosted Runner (Windows Server 2016, node.js 21)

Is this config is supported?

**Comment by Naimulnobel** [comment]
The issue you're facing might be due to the version of the `actions/setup-node` action you're using. The `actions/setup-node` action is used to set up a specific version of Node.js in your workflow. The latest version of this action is `v3`, and it supports Node.js versions up to `16.x`.

If you're trying to use Node.js `20` or `21`, it's likely that these versions are not supported by the `actions/setup-node` action. As of now, the latest stable version of Node.js is `16.x`, and the next version will be `17.x`.

If you're using a custom action and you want to use a specific version of Node.js, you should specify this version in the `package.json` file of your action. Then, when your action runs, it will use the version of Node.js specified in the `package.json` file, regardless of the version set up by the `actions/setup-node` action.

Here's an example of how you can specify the Node.js version in the `package.json` file:

```json
{
  "name": "my-action",
  "version": "1.0.0",
  "description": "My custom GitHub action",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "engines": {
    "node": "21.x"
  },
  "dependencies": {
    // Your dependencies here
  }
}
```

In this example, the `"engines"` field specifies that the action should use Node.js `21.x`. Please note that this will only work if Node.js `21.x` is installed on the runner where your action is running. If it's not installed, you'll need to install it manually.
